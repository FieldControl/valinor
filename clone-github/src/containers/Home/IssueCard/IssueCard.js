import React from "react";
import PropTypes from "prop-types";
import "./IssueCard.scss";
import eIcon from "../../../assets/e.png";

import { formatUrl } from "../../../helpers/format";

const IssueCard = ({
  repository_url,
  title,
  number,
  state,
  updated_at,
  user,
  comments,
  body,
}) => {
  return (
    <div className="issue__card">
      <a className="issue__card__link issue__card__icon" href="/">
        {" "}
        <img
          width="12px"
          height="12px"
          className="icon icon--issue"
          src={eIcon}
        />
        <div className="issue__header">
          <p className="issue__card__description ">
            {formatUrl(repository_url)}
          </p>
          <p className="issue__card__number">#{number}</p>
        </div>
      </a>

      <ul className="issue__card__content">
        <li className="issue__title">{title}</li>
        <li className="issue__body">{`${String(body).slice(0, 200)} ...`}</li>
        {user !== null ? (
          <li className="issue__card__user">{user.login.toUpperCase()}</li>
        ) : null}
        <li className="issue__card__icon">{state}ed</li>
        <li className="issue__card__updated">
          {new Date(updated_at).toLocaleDateString()}
        </li>
        {comments > 0 ? <li>{comments} comment</li> : null}
      </ul>
    </div>
  );
};

IssueCard.propTypes = {
  repository_url: PropTypes.string,
  title: PropTypes.string,
  updated_at: PropTypes.string,
  number: PropTypes.number,
  comments: PropTypes.number,
  user: PropTypes.object,
  state: PropTypes.bool,
  body: PropTypes.string,
};

export default IssueCard;
