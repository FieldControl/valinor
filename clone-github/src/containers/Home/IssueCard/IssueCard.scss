.issue__card {
  box-sizing: border-box;
  padding: 20px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: left;
  text-align: left;
  width: 100%;
  height: 100%;
  border-top: 1px solid silver;
}

.issue__card__link {
  font-family: AssistantRegular, serif;

  & em {
    font-weight: bold;
  }
  &:hover {
    opacity: 0.7;
  }
}
.issue__card__content {
  font-family: Lato, sans-serif;
  position: relative;
  box-sizing: border-box;
  padding: 1px;
  font-size: 12px;
  color: #586069;
  width: 100%;
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  & li {
    margin-top: 5px;
  }
}
.issue__card__description {
  position: relative;
  box-sizing: border-box;
  padding: 1px;
  font-size: 12px;
  font-weight: bold;
  margin-top: 5px;
}
.issue__card__number {
  box-sizing: border-box;
  padding: 1px;
  font-size: 12px;
  color: #666768;
  margin-top: 5px;
  margin-right: 5px;
}
.issue__header {
  box-sizing: border-box;
  position: relative;
  display: flex;
}
.issue__title {
  width: 100%;
  font-size: 1rem;
  color: #f92104;
  margin-right: 2px;
}
.issue__body {
  position: relative;
  box-sizing: border-box;
  font-size: 14px;
  color: rgb(22, 22, 22);
  line-height: 1.5;
  width: 100%;
}
.icon {
  width: 14px;
  height: 14px;
  margin-right: 5px;
  position: relative;
}
.icon--issue {
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  position: absolute;
  left: 0;
  margin-top: 5px;
}
.issue__card__updated {
  margin-right: 5px;
}
.issue__card__user {
  font-weight: bold;
  margin-right: 5px;
}
.issue__card__icon {
  margin-right: 5px;
  text-decoration: none;
}
.icon__circle {
  box-sizing: border-box;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  margin-right: 3px;
  margin-left: 3px;
  border: 2px solid black;
}

@media (max-width: 850px) {
  .issue__body {
    overflow: hidden;
  }
}
