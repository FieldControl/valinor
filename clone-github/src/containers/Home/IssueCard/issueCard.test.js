import { render, cleanup } from "@testing-library/react";
import IssueCard from "./IssueCard.js";

afterEach(cleanup);

test("should take a snapshoot", () => {
  const { asFragment } = render(<IssueCard />);
  expect(asFragment(<IssueCard />)).toMatchSnapshot();
});
