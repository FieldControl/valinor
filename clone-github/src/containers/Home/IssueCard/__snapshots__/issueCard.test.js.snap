// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshoot 1`] = `
<DocumentFragment>
  <div
    class="issue__card"
  >
    <a
      class="issue__card__link issue__card__icon"
    >
       
      <img
        class="icon icon--issue"
        height="12px"
        src="e.png"
        width="12px"
      />
      <div
        class="issue__header"
      >
        <p
          class="issue__card__number"
        >
          #
        </p>
      </div>
    </a>
    <ul
      class="issue__card__content"
    >
      <li
        class="issue__title"
      />
      <li
        class="issue__card__icon"
      >
        ed
      </li>
      <li
        class="issue__card__updated"
      >
        Invalid Date
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
